Examples/ObjectiveC/SymbolLayerZOrderExample.m

```diff
@@ -0,0 +1,106 @@
+//
+//  SymbolLayerZOrderExample.m
+//  Examples
+//
+//  Created by Sam Fader on 10/15/18.
+//  Copyright © 2018 Mapbox. All rights reserved.
+//
+
+#import "SymbolLayerZOrderExample.h"
+@import Mapbox;
+
+NSString *const MBXExampleSymbolLayerZOrder = @"SymbolLayerZOrderExample";
+
+@interface SymbolLayerZOrderExample () <MGLMapViewDelegate>
+
+@property (nonatomic) MGLMapView *mapView;
+@property (nonatomic) MGLSymbolStyleLayer *layer;
+
+@end
+
+@implementation SymbolLayerZOrderExample
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+    // Create a new map view using the Mapbox Light style.
+    self.mapView = [[MGLMapView alloc] initWithFrame:self.view.bounds
+                                            styleURL:[MGLStyle lightStyleURL]];
+    
+    self.mapView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
+    self.mapView.tintColor = [UIColor darkGrayColor];
+    
+    // Set the map’s center coordinate and zoom level.
+    self.mapView.centerCoordinate = CLLocationCoordinate2DMake(-41.292650,174.778768);
+    self.mapView.zoomLevel = 11.5;
+    
+    self.mapView.delegate = self;
+    [self.view addSubview: self.mapView];
+}
+
+// Wait until the style is loaded before modifying the map style.
+- (void)mapView:(MGLMapView *)mapView didFinishLoadingStyle:(MGLStyle *)style {
+    
+    // Add icons to the map's style.
+    [style setImage:([UIImage imageNamed:@"oval"]) forName:@"oval"];
+    [style setImage:([UIImage imageNamed:@"squircle"]) forName:@"squircle"];
+    [style setImage:([UIImage imageNamed:@"star"]) forName:@"star"];
```

Comment:
Just for clarity, I recommend making the string used for the image name something more unique so that it doesn't match, like this:

```objc
[style setImage:([UIImage imageNamed:@"oval-image"]) forName:@"oval"];
```

That way once you set up your dictionary it would illustrate that each pair of the key/value strings are different values.